packages/gatsby/cache-dir/loader.js

```diff
@@ -226,6 +230,54 @@ export class BaseLoader {
           // undefined if final result is an error
           return pageResources
         })
+
+        const staticQueryBatchPromise = Promise.all(
+          staticQueryHashes.map(staticQueryHash => {
+            console.log(
+              `${pagePath} needs a static query with hash ${staticQueryHash}`
+            )
+
+            // TODO: Get from cache
+
+            // TODO: Get in flight Promise
+
+            return doFetch(`/static/d/${staticQueryHash}.json`).then(req => {
+              const jsonPayload = JSON.parse(req.responseText)
+              console.log(
+                `Static query with hash ${staticQueryHash} was fetched`,
+                jsonPayload
+              )
+              // this.staticQueryDb.set(staticQueryHash, jsonPayload)
+              return { staticQueryHash, jsonPayload }
+            })
+          })
+        ).then(staticQueryResults => {
+          console.log({
+            staticQueryResults,
+          })
+          const staticQueryResultsMap = staticQueryResults.reduce(
+            (map, { staticQueryHash, jsonPayload }) => {
+              map[staticQueryHash] = jsonPayload
+              return map
+            },
+            {}
+          )
+          // emitter.emit(`onPostLoadStaticQueryResults`, {
+          //   staticQueryResultsMap,
+          // })
+          this.staticQueryDb.set(pagePath, staticQueryResultsMap)
+          return staticQueryResultsMap
+        })
+
+        return Promise.all([
```

Comment:
The double `Promise.all` looks awkward but I have no better suggestion, especially since you also rely on the `.then` receiving the two results separately anyways. :shrug: